lib/azure/classes.ts

```diff
@@ -0,0 +1,55 @@
+import { AzureadProvider } from '@cdktf/provider-azuread/lib/provider';
+import { AzurermProvider } from '@cdktf/provider-azurerm/lib/provider';
+import { AzurermBackend } from 'cdktf/lib/backends/azurerm-backend';
+import { Construct } from 'constructs';
+import { EnvProps } from './interfaces';
+
+export enum AzureProviders {
+  AZURE = 'azure_provider',
+  AZUREAD = 'azure_ad_provider'
+
+}
+
+export interface AzureConstructProps {
+  isLocalBackend?: boolean;
+  provider?: AzureProviders;
+  env: EnvProps;
+  tags?: Record<string, string>;
+}
+
+export class AzureConstruct extends Construct {
+  public readonly name: string;
+  public readonly tags: Record<string, string>;
+
+  constructor(scope: Construct, id: string, props: AzureConstructProps) {
+    super(scope, id);
+
+    const { isLocalBackend = false, provider = AzureProviders.AZURE, env: { name, env, location, resourceGroupName }, tags = {} } = props;
+
+    this.tags = { name, env, id, location, ...tags };
+    this.name = `${name}-${env}`;
+
+    switch (provider) {
+      case AzureProviders.AZURE:
+        new AzurermProvider(this, 'azure_provider', {
+          skipProviderRegistration: true,
+          features: {},
+        });
+        break;
+
+      case AzureProviders.AZUREAD:
+        new AzureadProvider(this, 'azure_ad_provider', {});
+        break;
+    }
+
+
+    if (!isLocalBackend) {
+      new AzurermBackend(this, {
+        resourceGroupName,
+        containerName: this.name,
+        key: `${name}.${env}.${id}.terraform.tfstate`,
```

Comment:
is it correct to get name from env, then use env? cause it inconsistent with other PRs https://github.com/atre/trn-components/pull/8/files#diff-acc1e5780158e5b0030484d77f18a2cf7f9dac5ec563924571cf9155b6a4539eR26

Reply:
This PR introduces env's so I've added it